PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/layers/data_link_layer.cpp

```diff
@@ -0,0 +1,93 @@
+#include "layers/data_link_layer.h"
+
+/**
+ * @brief Updates the shared credit limit values based on the received DLLP information
+ * @param flit Flit containing the DLLPs
+ * @param P_SHARED_CREDIT_LIMIT Array of shared credit limit for posted traffic
+ * @param NP_SHARED_CREDIT_LIMIT Array of shared credit limit for non-posted traffic
+ * @param CPL_SHARED_CREDIT_LIMIT Array of shared credit limit for completion traffic
+ * @param FI1 First indication (FI1) flag for credit update
+ * @param FI2 Second indication (FI2) flag for credit update
+*/
+void DataLinkLayer::updateCreditLimit(Flit flit, int& P_SHARED_CREDIT_LIMIT[2], int& NP_SHARED_CREDIT_LIMIT[2], int& CPL_SHARED_CREDIT_LIMIT[2], bool& FI1, bool& FI2) {
+	// Static variables to store the flags and state across function calls
+	static bool discard, pFc1Flag, npFc1Flag, cplFc1Flag, pFc2Flag, npFc2Flag, cplFc2Flag;
+
+	// The flit contains two DLLPs, dllp_1 = flit.dllpPayload[0:32], dllp_2 = flit.dllpPayload[32:]
+	auto dllpPayload = flit.dllpPayload.to_ulong();
+
+	// the two dllps found in the flit
+	Dllp Dllps[2] = { Dllp::DllpObjRep(std::bitset<32>((dllpPayload >> 32) & 0xffffffff)), Dllp::DllpObjRep(std::bitset<32>(dllpPayload & 0xffffffff)) };
```

Comment:
I'm not sure what this means. A flit has only one DLLP, which is found from bit 0 to 31
![image](https://user-images.githubusercontent.com/57875312/221368940-dcac584a-3351-403f-806a-e59fd2643a18.png)